### PR TITLE
Include hazelcast-mapstore jar in full distribution

### PR DESCRIPTION
**Pull request by frant-hartm**
Checklist:
- [x] Labels (`Team:`, `Type:`, `Source:`, `Module:`) and Milestone set
- [x] Label `Add to Release Notes` or `Not Release Notes content` set
- [x] Request reviewers if possible

### CODE REVIEW

**Review by olukas — CHANGES_REQUESTED** [review]
I don't see related jar in full distribution. When I call `mvn clean install -DskipTests` and then `unzip -l distribution/target/hazelcast-5.2-SNAPSHOT.zip | grep mapstore` it didn't find anything.

There is also following in the log of maven build:
```
[WARNING] The following patterns were never triggered in this artifact inclusion filter:
o  'com.hazelcast:hazelcast-mapstore'
```

I guess `com.hazelcast:com.hazelcast:hazelcast-mapstore` -> `com.hazelcast.jet:hazelcast-mapstore` could fix it.

File: extensions/mapstore/pom.xml [hunk 1]
```diff
@@ -56,15 +56,15 @@
...
+            <version>${project.parent.version}</version>
             <scope>provided</scope>
         </dependency>
 
         <!-- Test dependencies -->
         <dependency>
             <groupId>com.hazelcast</groupId>
             <artifactId>hazelcast-sql</artifactId>
-            <version>5.2-SNAPSHOT</version>
+            <version>${project.parent.version}</version>
```
Comment (ldziedziul) [inline extensions/mapstore/pom.xml:67]: ❤️

File: distribution/pom.xml [hunk 2]
```diff
@@ -394,6 +394,11 @@
             <artifactId>hazelcast-3-connector-common</artifactId>
             <version>${project.version}</version>
         </dependency>
+        <dependency>
+            <groupId>com.hazelcast.jet</groupId>
```
Comment (ldziedziul) [inline distribution/pom.xml:398]: Do we want to keep a separate group for jet-related artifects?

Comment (frant-hartm) [reply distribution/pom.xml]: Good point, changed to plain `com.hazelcast` for this module.

File: extensions/mapstore/pom.xml [hunk 3]
```diff
@@ -20,13 +20,16 @@
          xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
 
     <modelVersion>4.0.0</modelVersion>
+
+    <groupId>com.hazelcast</groupId>
```
Comment (ldziedziul) [inline extensions/mapstore/pom.xml:24]: Why do we need it? Other extensions declare only `artifactId`

Comment (frant-hartm) [reply extensions/mapstore/pom.xml:24]: The `groupId` is inherited from the parent, which is `com.hazelcast.jet` for the extensions. If we want to have `com.hazelcast` for the `groupId` we need to override it here.

Maybe I misunderstood what you wanted to do here https://github.com/hazelcast/hazelcast/pull/22213#discussion_r970704102. Can you clarify that? :-)

**Review by olukas — APPROVED** [review]

**Comment by frant-hartm** [comment]
run-lab-run

File: extensions/mapstore/pom.xml (hunk 3 above)
Comment (ldziedziul) [reply extensions/mapstore/pom.xml:24]: ah, all right, you got me right :)

**Comment by frant-hartm** [comment]
run-lab-run

**Comment by frant-hartm** [comment]
run-lab-run

**Pull request merged**